modules/hcp-bm-add-nodes-to-inventory.adoc

```diff
@@ -0,0 +1,310 @@
+----
+secret/pull-secret created
+----
+
+. Create the InfraEnv resource by adding the following content to a YAML file:
++
+[source,yaml]
+----
+apiVersion: agent-install.openshift.io/v1beta1
+kind: InfraEnv
+metadata:
+  name: hosted
+  namespace: <namespace_example>
+spec:
+  additionalNTPSources:
+  - <ip_address>
+  pullSecretRef:
+    name: pull-secret
+  sshAuthorizedKey: <ssh_public_key>
+----
+
+. Apply the changes to the YAML file by entering the following command:
++
+[source,terminal]
+----
+$ oc apply -f <infraenv_config>.yaml
+----
++
+Replace `<infraenv_config>` with the name of your file.
+
+. Verify that the `InfraEnv` resource was created by entering the following command:
++
+[source,terminal]
+----
+$ oc --kubeconfig ~/<directory_example>/mgmt-kubeconfig \
+  -n <namespace_example> get infraenv hosted
+----
+
+. Add bare-metal hosts by following one of two methods:
+
+** If you do not use the Metal3 Operator, obtain the discovery ISO from the `InfraEnv` resource and boot the hosts manually by completing the following steps:
+
+.. Download the live ISO by entering the following commands:
++
+[source,terminal]
+----
+$ oc get infraenv -A
+----
++
+[source,terminal]
+----
+$ oc get infraenv <namespace_example> -o jsonpath='{.status.isoDownloadURL}' -n <namespace_example> <iso_url>
+----
+
+.. Boot the ISO. The node communicates with the Assisted Service and registers as an agent in the same namespace as the `InfraEnv` resource.
+
+.. For each agent, set the installation disk ID and hostname, and approve it to indicate that the agent is ready for use. Enter the following commands:
++
+[source,terminal]
+----
+$ oc -n <hosted_control_plane_namespace> get agents
+----
++
+.Example output
+[source,terminal]
+----
+NAME                                   CLUSTER   APPROVED   ROLE          STAGE
+86f7ac75-4fc4-4b36-8130-40fa12602218                        auto-assign
+e57a637f-745b-496e-971d-1abbf03341ba                        auto-assign
+----
++
+[source,terminal]
+----
+$ oc -n <hosted_control_plane_namespace> \
+  patch agent 86f7ac75-4fc4-4b36-8130-40fa12602218 \
+  -p '{"spec":{"installation_disk_id":"/dev/sda","approved":true,"hostname":"worker-0.example.krnl.es"}}' \
+  --type merge
+----
++
+[source,terminal]
+----
+$ oc -n <hosted_control_plane_namespace> \
+  patch agent 23d0c614-2caa-43f5-b7d3-0b3564688baa -p \
+  '{"spec":{"installation_disk_id":"/dev/sda","approved":true,"hostname":"worker-1.example.krnl.es"}}' \
+  --type merge
+----
++
+[source,terminal]
+----
+$ oc -n <hosted_control_plane_namespace> get agents
+----
++
+.Example output
+[source,terminal]
+----
+NAME                                   CLUSTER   APPROVED   ROLE          STAGE
+86f7ac75-4fc4-4b36-8130-40fa12602218             true       auto-assign
+e57a637f-745b-496e-971d-1abbf03341ba             true       auto-assign
+----
+
+** If you use the Metal3 Operator, you can automate the creation of `BareMetalHost` objects by completing the following steps:
```

Comment:
This is not accurate:

"you can automate the creation of `BareMetalHost` objects by completing the following steps"

I would say something like:

"you can automate the bare metal host registration by creating the following objects"